routes/bots.js

```diff
@@ -5,104 +5,171 @@ const bot = require("../utils/discordbot");
-            (b.support && b.support.length > 2083) ||
-            (b.custominvite && b.custominvite.length > 2083) ||
-            !b.prefix || b.prefix.length > 15 ||
-            !b.shortdesc || b.shortdesc.length < 2 || b.shortdesc.length > 300 ||
-            (b.longdesc && b.longdesc.length > 10000) ||
-            !botTags.length || botTags.length > 6 || botTags.some(t => !allTags.includes(t))
+  const dBot = bot(config);
+  router.get("/", (req, res) => {
+    res.render("index", { title: "Bots" });
+  });
+
+  router.get("/add", (req, res) => {
+    if (!req.session.user) return res.redirect("/oauth2/login");
+    res.render("bots/add", { tags, title: "Adicionar Bot", libraries });
+  });
+
+  router.post("/add", async (req, res) => {
+    try {
+      if (!req.session.user) return res.redirect("/oauth2/login");
+      const b = req.body;
+      if (!(await captchaIsValid(config.recaptcha, b["g-recaptcha-response"])))
+        return res.render("message", {
+          message: "O Captcha precisa ser validado.",
+          url: req.originalUrl,
+        });
+      const allTags = Object.keys(tags);
+      const botTags = [
+        ...new Set(typeof b.tags == "string" ? [b.tags] : b.tags),
+      ];
+      const owners = b.owners ? [...b.owners] : null;
+      {
+        if (!owners || owners.some((o) => isNaN(o) || o.length != 18))
+          return res.render("message", {
+            message: "Lista de donos inválida.",
+            url: req.originalUrl,
+          });
+
+        if (isNaN(b.id) || b.id.length != 18)
+          return res.render("message", {
+            message: "ID do bot fornecido é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.library || !libraries.includes(b.library))
+          return res.render("message", {
+            message: "Biblioteca fornecida é inválida.",
+            url: req.originalUrl,
+          });
+        if (b.support && b.support.length > 2083)
+          return res.render("message", {
+            message: "Link do servidor de suporte é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.prefix || b.prefix.length > 15)
+          return res.render("message", {
+            message: "Prefixo do bot é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.shortdesc || b.shortdesc.length < 2 || b.shortdesc.length > 300)
+          return res.render("message", {
+            message: "Descrição curta é inválida.",
+            url: req.originalUrl,
+          });
+        if (b.longdesc && b.longdesc.length > 10000)
+          return res.render("message", {
+            message: "Descrição longa é inválida.",
+            url: req.originalUrl,
+          });
+        if (
+          !botTags.length ||
+          botTags.length > 6 ||
+          botTags.some((t) => !allTags.includes(t))
         )
-            return res.render("message", {message: "Algo de errado não está certo.", url: req.originalUrl});
```

Comment:
More specific errors to user know what happened and what didn't work

Reply:
We were doing this more specific check on the front end, but it really seems better to do both